akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1924,7 +1924,20 @@ private[stream] object Collect {
 
     def pushPull(): Unit =
       if (hasNext) {
-        push(out, currentIterator.next())
+        try {
+          val el = currentIterator.next()
+          println(s"el = ${el}")
+          push(out, el)
+        } catch {
+          case NonFatal(ex) ⇒ decider(ex) match {
+            case Supervision.Stop              ⇒ failStage(ex)
+            case Supervision.Resume if hasNext ⇒ pushPull()
+            case Supervision.Resume            ⇒ ??? // FIXME should get the next from f();  // if (!hasBeenPulled(in)) pull(in)
```

Comment:
Shouldn't be a `???` here right?

Reply:
see the fixme ;)

Reply:
Yeah, I just realized, how many iterators will gracefully continue after an exception?

Reply:
Yeah I wonder, we got a someone asking for it so figured I can try it. We'll see if their iterator actually does so